### PR TITLE
feat(components/lookup): add test harnesses to country field

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2195394](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2195394)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2822/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2822/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2822/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2822/playground/)

File: libs/components/lookup/testing/src/country-field/fixtures/country-field-harness-test.component.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  selector: 'sky-country-field-fixture',
+  templateUrl: './country-field-harness-test.component.html',
+  imports: [
+    FormsModule,
+    ReactiveFormsModule,
+    SkyCountryFieldModule,
+    SkyInputBoxModule,
+  ],
+})
+export class CountryFieldHarnessTestComponent {
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/lookup/testing/src/country-field/fixtures/country-field-harness-test.component.ts:23]: the two inputs below dont seem to be used in unit testing, can we remove them

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  public static override hostSelector = 'sky-country-field,.sky-input-box';
+
+  #documentRootLocator = this.documentRootLocatorFactory();
+
+  #getInput = this.locatorFor(SkyAutocompleteInputHarness);
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyCountryFieldHarness` that meets certain criteria.
+   */
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/lookup/testing/src/country-field/country-field-harness.ts:35]: Based on what we talked about at standup today, can we call out for consumers how to query country fields inside input boxes? This PR will most likely set the standard for wording it, but this is what I have in my datepicker code right now 
```These filters only work for standalone country fields. For country fields wrapped inside a `sky-input-box` place filters on the input box and query the country field using a `SkyInputBoxHarness`. See the code example for the input box implementation.```

Would need input from John Lyons on better wording though

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+ * Harness for interacting with a country field component in tests.
+ */
+export class SkyCountryFieldHarness extends SkyAutocompleteHarness {
+  /**
+   * Finds a standard country field component, or a country field component that is wrapped by an input box component.
+   * For input box implementations, we need to use the `.sky-input-box` selector since the `sky-country-field`
+   * element is removed from the DOM.
+   * @internal
+   */
+  public static override hostSelector = 'sky-country-field,.sky-input-box';
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/lookup/testing/src/country-field/country-field-harness.ts]: `.sky-input-box` as a selector also queries things that are not added in country field. To keep the queried harness similar between the two scenarios, can we change this to `.sky-country-field-container`?

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  public override async getValue(): Promise<string> {
+    return await super.getValue();
+  }
+
+  /**
+   * Gets the text displayed when no search results are found.
+   * This is always the default for country field.
+   * @internal
+   */
+  /* istanbul ignore next */
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/lookup/testing/src/country-field/country-field-harness.ts:144]: can we be more descriptive that this is being marked `@internal` to remove this value from the JSdocs for country field?

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+  /**
+   * Focuses the country field input.
+   */
+  public override async focus(): Promise<void> {
+    return await super.focus();
+  }
+
+  /**
+   * Gets the country field  `aria-labelledby` value.
+   * This is not needed for country field because the id is generated internally.
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/lookup/testing/src/country-field/country-field-harness.ts]: can we be more descriptive that this is being marked `@internal` to remove this value from the JSdocs for country field?

File: libs/components/lookup/testing/src/country-field/country-field-search-result-harness.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      HarnessPredicate.stringMatches(await harness.getText(), text),
+    );
+  }
+
+  /**
+   * Returns the value of the search result's descriptor property.
+   * This is not needed by country field because it is always set to the country name.
+   * @internal
+   */
+  public override async getDescriptorValue(): Promise<string> {
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/lookup/testing/src/country-field/country-field-search-result-harness.ts:32]: can we be more descriptive that this is being marked `@internal` to remove this value from the JSdocs for country field search results?

**Review by Blackbaud-SandhyaRajasabeson — COMMENTED** [review]
your PR title says `layout` when it should be `lookup`

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/lookup/testing/src/country-field/country-field-harness.ts:35]: Added

File: libs/components/lookup/testing/src/country-field/fixtures/country-field-harness-test.component.ts (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/lookup/testing/src/country-field/fixtures/country-field-harness-test.component.ts:23]: Removed

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts (hunk 3 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/lookup/testing/src/country-field/country-field-harness.ts]: Changed

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts (hunk 4 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/lookup/testing/src/country-field/country-field-harness.ts:144]: Updated

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts (hunk 5 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/lookup/testing/src/country-field/country-field-harness.ts]: Updated

File: libs/components/lookup/testing/src/country-field/country-field-search-result-harness.ts (hunk 6 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/lookup/testing/src/country-field/country-field-search-result-harness.ts:32]: Updated

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts (hunk 2 above)
Comment (Blackbaud-SandhyaRajasabeson) [reply libs/components/lookup/testing/src/country-field/country-field-harness.ts:35]: @blackbaud-johnly what do you think of this wording?

Comment (blackbaud-johnly) [reply libs/components/lookup/testing/src/country-field/country-field-harness.ts:35]: LGTM. I'd just suggest a couple small tweaks. I'll add a separate comment with those suggestions. Do you want me to also review other text in this PR?

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+  #documentRootLocator = this.documentRootLocatorFactory();
+
+  #getInput = this.locatorFor(SkyAutocompleteInputHarness);
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyCountryFieldHarness` that meets certain criteria.
+   * These filters only work for standalone country fields.
+   * For country fields wrapped inside a `sky-input-box` place filters
```
Comment (blackbaud-johnly) [inline libs/components/lookup/testing/src/country-field/country-field-harness.ts]:
```suggestion
   * For country fields wrapped inside `sky-input-box`, place filters
```

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+   * on the input box and query the country field using a `SkyInputBoxHarness`.
```
Comment (blackbaud-johnly) [inline libs/components/lookup/testing/src/country-field/country-field-harness.ts]:
```suggestion
   * on the input box instead, and query the country field using a `SkyInputBoxHarness`.
```

File: libs/components/lookup/testing/src/country-field/country-field-harness.ts [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+   * See the code example for the input box implementation.
```
Comment (blackbaud-johnly) [inline libs/components/lookup/testing/src/country-field/country-field-harness.ts]:
```suggestion
   * For the input box implementation, see the code example.
```

**Review by Blackbaud-SandhyaRajasabeson — APPROVED** [review]

**Pull request merged**